### PR TITLE
Feature/12891 pages security api

### PR DESCRIPTION
**Pull request by bost-h**
This PR adds the documentation for the pages and page security management API

### CODE REVIEW

File: source/includes/_page_security.md [hunk 1]
```diff
@@ -0,0 +1,473 @@
+# Pages security
+
+The page security is the set of rules that defines who (which users / groups) can explore or edit a given page.
+
+It is defined through 2 variables:
+
+* a page restriction through the `restricted` attribute in [the page object](#the-page-object), which define if a page is visible to every user who can explore the portal
```
Comment (juliendbg) [inline source/includes/_page_security.md]: `defines`

File: source/includes/_page_security.md [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+* specific rulesets for users and groups
+
+If the page is set as restricted, then the page will only appear in the portal for users who have a ruleset declared for them, either directly or through a group. Other users won't have any access to the page.
+Rulesets can also give users permission to edit the page and manage its security.
```
Comment (juliendbg) [inline source/includes/_page_security.md:12]: You can add another line break for this sentence to be in another paragraph.

File: source/includes/_page_security.md [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+
+### The page security API in a nutshell
```
Comment (juliendbg) [inline source/includes/_page_security.md]: I think you skipped a title level here (`##`), which may do strange things to the TOC generation, is it on purpose?

File: source/includes/_page_security.md [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+
+#### User level page security ruleset
+
+`/pages/{PAGE_SLUG}/security/users`
```
Comment (juliendbg) [inline source/includes/_page_security.md:18]: Do users know what a "slug" is? Do we use this term somewhere in the UI?

File: source/includes/_pages.md [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+    "author": {
+        "username": "john.doe"
+    }
+}
+```
+### Attributes
+
+Attribute | Description
+--------- | -----------
+`slug` <br> *string* | Human readable identifier used to generate the page's URL
```
Comment (juliendbg) [inline source/includes/_pages.md]: `the page URL`

File: source/includes/_pages.md [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+`layout` | The HTML layout to use for this page. Possible values are `single-main.html`, `main-left-col-right.html`, `homepage-searchbox.html`, `single-main-contact.html`, `dataset-catalog-overview.html`, `dataset-of-datasets.html`, `reuses.html`, `custom.html`
+`content` <br> *object* | Internationalized page content. The shape of the object depends on the chosen layout 
+`tags` <br> *array of strings*  | List of string used to classify and sort pages
+`restricted` <br> *boolean* | Defines if the page is visible to every user who can explore the portal
+`created_at` <br> *datetime*  | Date when the page was created
+`last_modified` <br> *datetime*  | Date when the page was last edited
+`last_modified_user` <br> *[user object](#the-user-object)* <br> <em class="expandable">expandable</em> | The user who last modified the page
+`author` <br> *[user object](#the-user-object)* <br> <em class="expandable">expandable</em> | The user who created the page
+`pushed_by_parent` <br> *boolean* | Inform if the page has been distributed by a parent domain.
+`has_subdomain_copies` <br> *boolean* | Inform if the page been distributed to any subdomain.
```
Comment (juliendbg) [inline source/includes/_pages.md]: I think we don't end this kind of sentences by dots in the rest of the doc

File: source/includes/_pages.md [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+}
+```
+### Attributes
+
+Attribute | Description
+--------- | -----------
+`slug` <br> *string* | Human readable identifier used to generate the page's URL
+`title` <br> *object* | Internationalized page title
+`description` <br> *string* | Page description
+`layout` | The HTML layout to use for this page. Possible values are `single-main.html`, `main-left-col-right.html`, `homepage-searchbox.html`, `single-main-contact.html`, `dataset-catalog-overview.html`, `dataset-of-datasets.html`, `reuses.html`, `custom.html`
```
Comment (juliendbg) [inline source/includes/_pages.md]: - `The`

File: source/includes/_pages.md [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+
+This endpoint returns a paginated list of all the pages that can be edited by this user.
+
+### Parameters
+
+Parameter | Default | Description
+--------- | ------- | -----------
+`search` <br> *string* | None | Performs a full text search on the `title`, `slug`, `description` and `tags` attributes to filter the pages
+`rows` <br> *string* | 10 | Number of items to return per page. Max value: 100
+`page` <br> *string* | 1 | The list page to return
```
Comment (juliendbg) [inline source/includes/_pages.md]: `The page to return`

File: source/includes/_pages.md [hunk 9]
```diff
@@ -0,0 +1,453 @@
...
+`sort` <br> *string* | None | Field on which to sort the results list
```
Comment (juliendbg) [inline source/includes/_pages.md]: `Field on which to sort the returned objects`

File: source/includes/_page_security.md (hunk 3 above)
Comment (bost-h) [reply source/includes/_page_security.md]: I used the same levels as Dataset security does, and TOC generation seems fine, but I agree it does not feel ideal, i'll change this

File: source/includes/_page_security.md (hunk 4 above)
Comment (bost-h) [reply source/includes/_page_security.md:18]: I don't think we use this term in the UI: In the UI we have "Page URL" label but the form input for the slug is prefixed by `/pages/` so it is more accurate.

In the Pages section, when I describe the Page object, I define the slug as "Human readable identifier used to generate the page's URL". Moreover, IMO an API documentation is for developers, and slug is a widely used term, almost official.

Do you still think I should document this more @juliendbg ?

Comment (juliendbg) [reply source/includes/_page_security.md:18]: No that's fine like that 👍

**Review by juliendbg — APPROVED** [review]
We should merge this @bost-h, it looks like it was forgotten.

**Comment by bost-h** [comment]
Should still be valid, I don't remember any changes to the API 👍

File: source/includes/_pages.md [hunk 10]
```diff
@@ -0,0 +1,453 @@
...
+### Attributes
+
+Attribute | Description
+--------- | -----------
+`slug` <br> *string* | Human readable identifier used to generate the page URL
+`title` <br> *object* | Internationalized page title
+`description` <br> *string* | Page description
+`layout` | The HTML layout to use for this page. Possible values are `single-main.html`, `main-left-col-right.html`, `homepage-searchbox.html`, `single-main-contact.html`, `dataset-catalog-overview.html`, `dataset-of-datasets.html`, `reuses.html`, `custom.html`
+`content` <br> *object* | Internationalized page content. The shape of the object depends on the chosen layout 
+`tags` <br> *array of strings*  | List of string used to classify and sort pages
```
Comment (meganelacheny) [inline source/includes/_pages.md]:
```suggestion
`tags` <br> *array of strings*  | List of strings used to classify and sort pages
```

File: source/includes/_pages.md [hunk 11]
```diff
@@ -0,0 +1,453 @@
...
+### Body
+
+Parameter | Description
+--------- | -----------
+`slug` <br> *string* | Human readable identifier used to generate the page's URL
+`title` <br> *object* | Internationalized page title
+`description` <br> *string* | Page description
+`layout` | The HTML layout to use for this page. Possible values are `single-main.html`, `main-left-col-right.html`, `homepage-searchbox.html`, `single-main-contact.html`, `dataset-catalog-overview.html`, `dataset-of-datasets.html`, `reuses.html`, `custom.html`
+`content` <br> *object* | Internationalized page content. The shape of the object depends on the chosen layout 
+`tags` <br> *array of strings*  | List of string used to classify and sort pages
```
Comment (meganelacheny) [inline source/includes/_pages.md]:
```suggestion
`tags` <br> *array of strings*  | List of strings used to classify and sort pages
```

File: source/includes/_pages.md [hunk 12]
```diff
@@ -0,0 +1,453 @@
...
+
+### Body
+
+Parameter | Description
+--------- | -----------
+`title` <br> *object* | Internationalized page title
+`description` <br> *string* | Page description
+`layout` | The HTML layout to use for this page. Possible values are `single-main.html`, `main-left-col-right.html`, `homepage-searchbox.html`, `single-main-contact.html`, `dataset-catalog-overview.html`, `dataset-of-datasets.html`, `reuses.html`, `custom.html`
+`content` <br> *object* | Internationalized page content. The shape of the object depends on the chosen layout 
+`tags` <br> *array of strings*  | List of string used to classify and sort pages
```
Comment (meganelacheny) [inline source/includes/_pages.md]:
```suggestion
`tags` <br> *array of strings*  | List of strings used to classify and sort pages
```

**Review by meganelacheny — APPROVED** [review]

**Pull request merged**